### PR TITLE
Massive generation of refutations via polynomials over Z/nZ

### PR DESCRIPTION
**Pull request by nomeata**
This PR adds a large number of automatically generated refutations by @carlini in https://github.com/carlini/equational_theories. All of them are magmas over Z/nZ with a polynomial of degree ≤2 as the operations.

For now, until we know better what to actually run, I excluded all magmas with 6 or more elements, just to keep this PR sizable.

### CODE REVIEW

File: equational_theories/FinitePoly/FactsSyntax.lean [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+others”.
+-/
+
+-- May ways to skin the cat here; using syntactic macros to expand the conjunction,
+-- or an actual predicate `Facts G : List Nat → List Nat → Prop`.
+
+-- Trying the syntactic variant here
+
+
+syntax "Facts " term:max " [" num,* "] " " [" num,* "]" : term
```
Comment (eric-wieser) [inline equational_theories/FinitePoly/FactsSyntax.lean:15]: Can you add a docstring here?

Comment (nomeata) [reply equational_theories/FinitePoly/FactsSyntax.lean:16]: Sure!

**Comment by nomeata** [comment]
Hmm, the blueprint CI broke. I think I’ll wait for the overall discussion around file structure to settle before continueing here. Anyone is welcome to pick it up though.

File: DecideBang.lean [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import Lean
+
+/-!
+This is a variant of the `decide` tactic that doesn’t actually check the proposition in meta code.
+This is useful in non-interactive mode, where we know it will succeed, and we really just want the
+check done once, in the kernel.
+
+This also includes a hack to do large disjunctions outside the `decdies`, as inference can blow up.
```
Comment (dwrensha) [inline DecideBang.lean:8]: decdies -> decides

File: DecideBang.lean [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      | _ => throwError "Expected Fin n equality"
+    | _ => throwError "Unsupported propositoin"
+
+
+/-!
+Like `decide!`, but only supports goals that are conjunctions of (possibly negations of) goals
+of the form `∀ (x … z : Fin n), lhs = rhs`.
+
+Using type class synthesis to infer the decidability instances can be very slow, slower than the
+actual proof checking, so this tactic construts the instances very directly.
```
Comment (dwrensha) [inline DecideBang.lean:74]: construts -> contructs

**Comment by nomeata** [comment]
Should be merged after https://github.com/teorth/equational_theories/pull/110 only.

**Pull request merged**